### PR TITLE
Fix/analysis module bugs

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: app/components/EntryCommentWrapper/CommentContext.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createContext } from 'react';
```
Comment (tnagorra) [inline app/components/EntryCommentWrapper/CommentContext.ts]: Let's move EntryCommentWrapper from component

File: app/views/Project/Tagging/Sources/LeadEditModal/index.tsx [hunk 2]
```diff
@@ -454,7 +455,8 @@ function LeadEditModal(props: Props) {
 
     const handleLeadChange = useCallback((newValue: SetValueArg<PartialFormType>) => {
         setValue(newValue, true);
-    }, [setValue]);
+        setPristine(false);
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/LeadEditModal/index.tsx]: Let's use toggle-form@beta

File: app/views/Project/Tagging/Sources/EntriesGrid/index.tsx [hunk 3]
```diff
@@ -174,6 +192,16 @@ function SourcesGrid(props: Props) {
         filters: rawFilters,
     } = props;
 
+    const [
+        commentsCountMap,
+        setCommentsCountMap,
+    ] = useState<{ [key in string]: number } | undefined>(undefined);
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/EntriesGrid/index.tsx]:
```suggestion
    ] = useState<{ [key in string]: number | undefined } | undefined>(undefined);
```

File: app/views/Project/Tagging/Export/utils.ts [hunk 4]
```diff
@@ -78,7 +78,7 @@ function transformLevelsColumnFirst(
 ): Level[] {
     const sectorFirstLevels = matrix2dProperties.columns.map((column) => {
         let sublevels: Level[] = [];
-        if (column.subColumns) {
+        if (column.subColumns.length > 0) {
```
Comment (tnagorra) [inline app/views/Project/Tagging/Export/utils.ts:82]: Let's add a note

File: app/views/Project/EntryEdit/index.tsx [hunk 5]
```diff
@@ -942,28 +1006,15 @@ function EntryEdit(props: Props) {
                             </Button>
                             <Button
                                 name={undefined}
-                                disabled={formPristine || !!selectedEntry}
+                                // NOTE: Enable when lead form is edited
+                                disabled={(
+                                    !formStale && leadValue.status === 'TAGGED'
+                                ) || !!selectedEntry}
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx]: This is still a bit complicated

File: app/views/Project/EntryEdit/index.tsx [hunk 6]
```diff
@@ -243,7 +263,7 @@ function EntryEdit(props: Props) {
     } = useForm(schema, defaultFormValues);
 
     const formStale = formValue?.entries?.some((entry) => entry.stale) ?? false;
-    const formPristine = !formStale;
+    const formPristine = !formStale && leadPristine;
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx]: Let's rename formStale to entriesStale

File: app/views/Project/EntryEdit/SourceDetails/index.tsx [hunk 7]
```diff
@@ -77,13 +80,20 @@ function SourceDetails(props: Props) {
...
     const {
         loading: leadOptionsLoading,
         data: leadOptions,
     } = useQuery<LeadOptionsQuery, LeadOptionsQueryVariables>(
         LEAD_OPTIONS,
     );
 
+    const handleLeadChange = useCallback((newValue: SetBaseValueArg<PartialFormType>) => {
+        setValue(newValue);
+        setPristine(false);
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/SourceDetails/index.tsx]: Let's use new toggle-form

File: app/views/ExploreDeep/index.tsx [hunk 8]
```diff
@@ -116,90 +119,105 @@ function ExploreDeep(props: Props) {
...
                     className={styles.rightContainer}
-                    heading="Top 5 most active project"
-                    headingDescription="Last 3 months"
-                    headingSize="medium"
-                    spacing="none"
                     headerClassName={styles.header}
                     headingContainerClassName={styles.heading}
+                    contentClassName={styles.projectListContainer}
+                    spacing="none"
+                    heading="Top 5 most active project"
```
Comment (tnagorra) [inline app/views/ExploreDeep/index.tsx]:
```suggestion
                    heading="Top 5 most active projectzzz"
```

File: app/views/ExploreDeep/index.tsx [hunk 9]
```diff
@@ -116,90 +119,105 @@ function ExploreDeep(props: Props) {
...
+                                valuePrecision={2}
+                                value={data?.projectExploreStats?.generatedExportsMonthly ?? 0}
+                            />
+                        </Card>
+                    </div>
+                    <div className={styles.playbackFrameworkContainer}>
+                        <Card className={styles.playbackCard}>
+                            <Message message="DEEP tutorials go here" />
+                        </Card>
+                        <ContainerCard
```
Comment (subinasr) [inline app/views/ExploreDeep/index.tsx:187]: This container height should match with the 'active project' container height.

File: app/components/entryReview/EntryComments/styles.css [hunk 10]
```diff
@@ -1,12 +1,14 @@
 .comment-button {
+    position: relative;
     overflow: visible;
 
     .comment-count {
         position: absolute;
-        top: 0;
+        top: -1em;
```
Comment (tnagorra) [inline app/components/entryReview/EntryComments/styles.css:8]: Let's add a note

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by subinasr — CHANGES_REQUESTED** [review]

**Comment by subinasr** [comment]
In case of no entries, the 'Finalize' button is not working.
Also. please check the DeepScan issues.

File: app/views/Project/EntryEdit/index.tsx [hunk 11]
```diff
@@ -242,8 +262,8 @@ function EntryEdit(props: Props) {
         error: formError,
     } = useForm(schema, defaultFormValues);
 
-    const formStale = formValue?.entries?.some((entry) => entry.stale) ?? false;
-    const formPristine = !formStale;
+    const entriesFormStale = formValue?.entries?.some((entry) => entry.stale) ?? false;
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx]: Let's memoize this

File: app/views/Project/EntryEdit/index.tsx [hunk 12]
```diff
@@ -293,6 +313,38 @@ function EntryEdit(props: Props) {
...
+    const handleLeadSave = useCallback((finalized: boolean) => {
+        if (!projectId || (!finalized && leadPristine)) {
+            return;
+        }
+        setIsFinalizeClicked(false);
+        const submit = createSubmitHandler(
+            leadFormValidate,
+            setLeadError,
+            (val) => {
+                const data = { ...val } as LeadInputType;
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx]: Any reason to copy val?

File: app/views/Project/Tagging/Sources/LeadEditModal/index.tsx [hunk 13]
```diff
@@ -467,7 +467,7 @@ function LeadEditModal(props: Props) {
                     name="save"
                     // FIXME: Add disabled during pristine later
                     // disabled={pristine || pending}
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/LeadEditModal/index.tsx]: Let's remove this comment as this is obsolete

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**